pkgs/development/python-modules/flask-restful/default.nix

```
@@ -30,7 +31,15 @@ buildPythonPackage rec {
   # conditional so that overrides are easier for web applications
   patches =
     lib.optionals (lib.versionAtLeast werkzeug.version "2.1.0") [ ./werkzeug-2.1.0-compat.patch ]
-    ++ lib.optionals (lib.versionAtLeast flask.version "3.0.0") [ ./flask-3.0-compat.patch ];
+    ++ lib.optionals (lib.versionAtLeast flask.version "3.0.0") [ ./flask-3.0-compat.patch ]
+    ++ [
+      # replace use nose by pytest: https://github.com/flask-restful/flask-restful/pull/970
+      (fetchpatch2 {
+        url = "https://github.com/flask-restful/flask-restful/commit/6cc4b057e5450e0c84b3ee5f6f7a97e648a816d6.patch?full_index=1";
```

```suggestion
        url = "https://github.com/flask-restful/flask-restful/commit/6cc4b057e5450e0c84b3ee5f6f7a97e648a816d6.patch";
```
or is there a reason to use the full index? (If you decide to change this, please remember to change the `hash` as well)

---

@emilazy wanted me to use them

---

See https://github.com/NixOS/nixpkgs/issues/266556; GitHub patches are not reproducible once repos grow big enough for the hash lengths to change, and this has actually bit us. This isn’t a problem on most repositories but I’d still consider it best practice to use `?full_index=1` until we have a `fetchpatch3` that handles this correctly. (It’s on my infinite to‐do list…)

Sorry for not reviewing these nose PRs for a few days. I will get back to it soon.